src/backend/marsha/core/api/video.py

```diff
@@ -104,6 +104,17 @@ def get_serializer_context(self):
         if isinstance(user, TokenUser):
             context.update(user.token.payload)
 
+        # If the user is a JWT token and has roles check if roles are admin or instructor
+        if (
+            isinstance(user, TokenUser)
+            and user.token.get("roles")
+            and (
+                bool(LTI_ROLES[ADMINISTRATOR] & set(user.token.get("roles")))
+                or bool(LTI_ROLES[INSTRUCTOR] & set(user.token.get("roles")))
+            )
+        ):
+            context["is_admin"] = True
+
```

Comment:
Can we use `IsTokenInstructor` and `IsTokenAdmin` from permissions.py here ?
Also maybe simplify a bit like this :
```suggestion
        if isinstance(user, TokenUser):
            context.update(user.token.payload)

            # If the user is a JWT token and has roles check if roles are admin or instructor
            context["is_admin"] =  (
                user.token.get("roles")
                and (
                    bool(LTI_ROLES[ADMINISTRATOR] & set(user.token.get("roles")))
                    or bool(LTI_ROLES[INSTRUCTOR] & set(user.token.get("roles")))
                )
            )
```

Reply:
But this code is not related to permissions

Reply:
In fact it can be used yes, I will do it.